java/code/src/com/suse/manager/maintenance/MaintenanceManager.java

```diff
@@ -695,6 +696,71 @@ public boolean isActionInMaintenanceWindow(Action action, MaintenanceSchedule sc
         return false;
     }
 
+    private Collection<CalendarComponent> getCalendarForNow(MaintenanceSchedule ms) {
+        return getScheduleEventsAtDate(new Date(), ms, parseCalendar(ms.getCalendarOpt().get()));
+    }
+
+    /**
+     * Check if system is in maintenance mode
+     *
+     * @param server the server to check
+     * @return true when the action is inside of a maintenance window, otherwise falsegg
+     */
+    public static boolean checkIfInMaintenanceMode(MinionServer server) {
+        if (server.getMaintenanceScheduleOpt().isEmpty()) {
+            return true;
+        }
+        else {
+            MaintenanceManager mm = MaintenanceManager.instance();
+            Collection<CalendarComponent> cc = mm.getCalendarForNow(server.getMaintenanceScheduleOpt().get());
+            return !cc.isEmpty();
+        }
+    }
+
+    /**
+     * Log the number of servers skipped and if debugging is enabled list the server names
+     *
+     * @param servers the list of servers to log
+     */
+    public static void logSkippedMinions(List<MinionServer> servers) {
+       log.warn("Skipping action for " + servers.size() + " minions.");
+       if (log.isDebugEnabled()) {
+          String serverNames = servers.stream()
+          .map(m -> m.getName())
```

Comment:
Do you want to log names or ids? The message says ids, but you're logging names.

Reply:
well names would make more sense in this case?  as it's what the people will see

Reply:
Yes, they are more readable, but they are not unique. The readability is not that important here anyway (it's a log). So I think ids are a better option here.